diff --git a/node_modules/@react-pdf/font/lib/index.browser.js b/node_modules/@react-pdf/font/lib/index.browser.js
index c50b03a..a616200 100644
--- a/node_modules/@react-pdf/font/lib/index.browser.js
+++ b/node_modules/@react-pdf/font/lib/index.browser.js
@@ -56,7 +56,7 @@ var FontSource = /*#__PURE__*/function () {
     this.src = src;
     this.fontFamily = fontFamily;
     this.fontStyle = fontStyle || 'normal';
-    this.fontWeight = 400;
+    this.fontWeight = fontWeight || 400;
     this.data = null;
     this.loading = false;
     this.options = options;
@@ -150,7 +150,7 @@ var Font = /*#__PURE__*/function () {
         fontStyle = _ref2.fontStyle,
         options = _objectWithoutPropertiesLoose(_ref2, ["src", "fontWeight", "fontStyle"]);
 
-    this.sources.push(new FontSource(src, this.fontFamily, fontStyle, fontWeight, options));
+    this.sources.push(new FontSource(src, this.family, fontStyle, fontWeight, options));
   };
 
   _proto2.resolve = function resolve(descriptor) {
diff --git a/node_modules/@react-pdf/font/lib/index.js b/node_modules/@react-pdf/font/lib/index.js
index 4916413..1e788b0 100644
--- a/node_modules/@react-pdf/font/lib/index.js
+++ b/node_modules/@react-pdf/font/lib/index.js
@@ -28,7 +28,7 @@ class FontSource {
     this.src = src;
     this.fontFamily = fontFamily;
     this.fontStyle = fontStyle || 'normal';
-    this.fontWeight = 400;
+    this.fontWeight = fontWeight || 400;
     this.data = null;
     this.loading = false;
     this.options = options;
