diff --git a/node_modules/electron-publish/out/publisher.js b/node_modules/electron-publish/out/publisher.js
index 2358e2a..8dc5b23 100644
--- a/node_modules/electron-publish/out/publisher.js
+++ b/node_modules/electron-publish/out/publisher.js
@@ -66,8 +66,15 @@ class HttpPublisher extends Publisher {
 }
 exports.HttpPublisher = HttpPublisher;
 function getCiTag() {
-    const tag = process.env.TRAVIS_TAG || process.env.APPVEYOR_REPO_TAG_NAME || process.env.CIRCLE_TAG || process.env.BITRISE_GIT_TAG || process.env.CI_BUILD_TAG || process.env.BITBUCKET_TAG;
+    const tag = getGithubCiTag() || process.env.TRAVIS_TAG || process.env.APPVEYOR_REPO_TAG_NAME || process.env.CIRCLE_TAG || process.env.BITRISE_GIT_TAG || process.env.CI_BUILD_TAG || process.env.BITBUCKET_TAG;
     return tag != null && tag.length > 0 ? tag : null;
 }
+
+function getGithubCiTag() {
+    const ref = process.env.GITHUB_CI_REF
+    if (!ref) return null
+    return ref.split('/tags/')[1]
+}
+
 exports.getCiTag = getCiTag;
 //# sourceMappingURL=publisher.js.map
\ No newline at end of file
